BLE Indoor Positioning/src/main/java/com/nexenio/bleindoorpositioning/location/distance/BeaconDistanceCalculator.java

```diff
@@ -23,6 +23,22 @@ public static float calculateDistanceTo(Beacon beacon) {
         return calculateDistanceTo(beacon, beacon.getFilteredRssi());
     }
 
+    /**
+     * Use Pythagoras with altitude to calculate more accurate distance to the beacon, if the
+     * distance is larger than the altitude.
+     */
+    public static float calculateDistanceWithAltitudeTo(Beacon beacon, float rssi) {
```

Comment:
Technically, this compensates for the altitude delta between device and beacon. I find the name `...WithAltitude` misleading, as it will return a distance without the altitude delta.